### PR TITLE
Add sort window impls

### PR DESCRIPTION
**Pull request by LahiruJayasekara**

### CODE REVIEW

File: stdlib/streams/src/main/ballerina/streams/windows.bal [hunk 1]
```diff
@@ -1539,3 +1539,296 @@ public function delayWindow(any[] windowParameters, function (StreamEvent[])? ne
...
+        self.sortedWindow = new;
+        self.lengthToKeep = 0;
+        self.sortMetadata = [];
+        self.fields = [];
+        self.sortTypes = [];
+        self.initParameters(windowParameters);
+    }
+
+    public function initParameters(any[] parameters) {
+        if(parameters.length() >= 3 && parameters.length() % 2 == 1) {
```
Comment (mohanvive) [inline stdlib/streams/src/main/ballerina/streams/windows.bal]: Can u invert the 'if' condition and follow fail fast approach. ?
```suggestion
        if(!(parameters.length() >= 3 && parameters.length() % 2 == 1)) {
        error err = error("Sort window should have two or more " +
                 "parameters (<int> windowLength, <string> attribute1, <string> order1, " +
                 "<string> attribute2, <string> order2, ...), but found " + parameters.length()
                 + " input attributes" );
             panic err;
             }
```

File: stdlib/streams/src/main/ballerina/streams/windows.bal [hunk 2]
```diff
@@ -1539,3 +1539,296 @@ public function delayWindow(any[] windowParameters, function (StreamEvent[])? ne
...
+            }
+
+            int i = 1;
+            while(i < parameters.length()) {
+                any nextParameter = parameters[i];
+                if(nextParameter is string) {
+                    if (i % 2 == 1) {
+                        self.fields[self.fields.length()] = nextParameter;
+                    } else {
+                        if (nextParameter == "ascending" || nextParameter == "descending") {
```
Comment (mohanvive) [inline stdlib/streams/src/main/ballerina/streams/windows.bal]: "ascending" and "descending" - move them to constants. I think we already have them as constants.

File: stdlib/streams/src/main/ballerina/streams/windows.bal [hunk 3]
```diff
@@ -1539,3 +1539,296 @@ public function delayWindow(any[] windowParameters, function (StreamEvent[])? ne
...
+            if (y is int) {
+                return x < (<float>y) ? -1 : x == <float>y ? 0 : 1;
+            }
+            else {
+                return x < y ? -1 : x == y ? 0 : 1;
+            }
+        }
+    }
+
+    public function stringSort(string x, string y) returns int {
```
Comment (mohanvive) [inline stdlib/streams/src/main/ballerina/streams/windows.bal]: Is it a new sort function? Can we reuse what we implemented for orderby?

**Comment by LahiruJayasekara** [comment]
closing and sending the PR to release-0.990.0 branch

**Pull request closed without merging**